### PR TITLE
Change email sender to include name; refactoring of email module; unit tests

### PR DESCRIPTION
**Pull request by vertexian**
The primary thing I wanted to do here was to change the from header on outgoing emails to 'Gig-o-Matic <gigomatic.superuser@gmail.com>', since it's been bugging me for a while that it's a raw email address without a sender name. The refactoring and unit tests happened 'cause I can't help myself with redundant code.

### CODE REVIEW

File: member.py [hunk 1]
```diff
@@ -190,6 +189,7 @@ def get_add_gig_band_list(cls, req, the_member_key):
 
     @classmethod
     def get_forums(cls, req, the_member_key):
+        import forum
```
Comment (vertexian) [inline member.py]: This bit is the one thing I haven't tested that I'm pretty sure is safe but not certain. For some reason running the unit tests (i.e. with a __main__ entry point) complains about circular dependencies, when loading the app from dev_appserver or the production App Engine does not.

Comment (aaronoppenheimer) [inline member.py]: The forum stuff is pretty half-baked. We just comment out anything giving us trouble.

File: goemail.py [hunk 2]
```diff
@@ -1,137 +1,68 @@
...
 import re
-
 import pickle
 
 from webapp2_extras import i18n
-from webapp2_extras import jinja2
 from webapp2_extras.i18n import gettext as _
 
-SENDER_EMAIL = 'gigomatic.superuser@gmail.com'
+_admin_email_address = 'Gig-o-Matic <gigomatic.superuser@gmail.com>'
```
Comment (aaronoppenheimer) [inline goemail.py]: Huh - when I try this in the dev environment, it crashes:

SMTPSenderRefused: (550, "5.1.0 <Gig-o-Matic <gigomatic.superuser@gmail.com>> size=698 invalid address 'Gig-o-Matic <gigomatic.superuser@gmail.com>'", <email.header.Header instance at 0x10c999b48>)

Comment (aaronoppenheimer) [inline goemail.py]: Oh, the comment dropped the brackets. Is:

`SMTPSenderRefused(550, "5.1.0 <Gig-o-Matic <gigomatic.superuser@gmail.com>> size=2739 invalid address 'Gig-o-Matic <gigomatic.superuser@gmail.com>'", <email.header.Header instance at 0x10d3c2200>)`

Comment (vertexian) [reply goemail.py]: I was afraid of that. The development mock for SMTP doesn't test this of course, so I had to guess what format it wanted. I'll see if I can find docs and tweak accordingly.

**Comment by vertexian** [comment]
Merge was tricky, but I think I got it right, and the mail unit tests pass.

I was able to reproduce the failure you see in the dev environment. It's actually a bug in how the mail stub works in GAE. The sender address is used for both the From header and the SMTP "MAIL FROM" command, where the latter _must_ be a raw email address. Production GAE shouldn't have this limitation, [as evidenced by their docs](https://cloud.google.com/appengine/docs/standard/python/mail/sending-mail-with-mail-api#sending_mail_with_mailsend_mail). The conditional statement I used is also from the docs. I don't have the ability to test the production case, but it's as safe as I can make it.

What do you think?

**Pull request merged**